docker/Dockerfile

```diff
@@ -144,6 +157,32 @@ RUN --mount=type=ssh \
   && pip uninstall -y ansible ansible-core \
   && /autoware/cleanup_apt.sh
 
+# Install rosdep dependencies
+COPY --from=rosdep-depend /rosdep-core-common-depend-packages.txt /tmp/rosdep-core-common-depend-packages.txt
+# hadolint ignore=SC2002
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+  apt-get update \
+  && cat /tmp/rosdep-core-common-depend-packages.txt | xargs apt-get install -y --no-install-recommends \
+  && /autoware/cleanup_apt.sh
+
+RUN --mount=type=cache,target=${CCACHE_DIR} \
+  --mount=type=bind,from=rosdep-depend,source=/autoware/src/core/autoware_adapi_msgs,target=/autoware/src/core/autoware_adapi_msgs \
+  --mount=type=bind,from=rosdep-depend,source=/autoware/src/core/autoware_cmake,target=/autoware/src/core/autoware_cmake \
+  --mount=type=bind,from=rosdep-depend,source=/autoware/src/core/autoware_internal_msgs,target=/autoware/src/core/autoware_internal_msgs \
+  --mount=type=bind,from=rosdep-depend,source=/autoware/src/core/autoware_lanelet2_extension,target=/autoware/src/core/autoware_lanelet2_extension \
+  --mount=type=bind,from=rosdep-depend,source=/autoware/src/core/autoware_msgs,target=/autoware/src/core/autoware_msgs \
+  --mount=type=bind,from=rosdep-depend,source=/autoware/src/core/autoware_utils,target=/autoware/src/core/autoware_utils \
```

Comment:
Is it possible to copy the whole directory under `/autoware/src/core` and remove the `/autoware/src/core/autoware.core` instead of specifying individual folders? That way, users won't have to update the Dockerfile when they would like to add new dependent repository to autoware.repos file.

Reply:
Unfortunately, regular expressions cannot be used with the `mount` option.
It is possible to handle https://github.com/autowarefoundation/autoware/blob/main/docker/Dockerfile#L27 by creating an intermediate stage that copies `src/core` and then removes only `src/core/autoware.core`, but this approach would be very tricky.

Reply:
If we create a `core/external` directory similar to the `universe/external` directory and move everything except `autoware.core` there, it seems like individual specifications will no longer be necessary.
https://github.com/autowarefoundation/autoware/blob/main/docker/Dockerfile#L35

Reply:
I see. Having external directory might confuse the existing users when they try to update the workspace.

Maybe we can merge as is for now, and reconsider splitting the directory when we start to add more repositories under autoware.repos